src/dag-node.js

```diff
@@ -186,17 +230,25 @@ module.exports = class DAGNode {
     return this
   }
 
-  toJSON () {
-    return {
-      Data: this.data,
-      Links: this.links.map((l) => l.toJSON()),
-      Hash: mh.toB58String(this.multihash()),
-      Size: this.size()
-    }
+  toJSON (callback) {
+    parallel([
+      (cb) => this.size(cb),
+      (cb) => this.multihash(cb)
+    ], (err, results) => {
+      if (err) {
+        return callback(err)
+      }
+
+      callback(null, {
+        Data: this.data,
+        Links: this.links.map((l) => l.toJSON()),
+        Hash: mh.toB58String(results[1]),
+        Size: results[0]
+      })
+    })
   }
 
   toString () {
-    const hash = mh.toB58String(this.multihash())
-    return `DAGNode <${hash} - data: "${this.data.toString()}", links: ${this.links.length}, size: ${this.size()}>`
+    return `DAGNode <data: "${this.data.toString()}", links: ${this.links.length}>`
```

Comment:
The alternative is to make `toString` async (to be able to fetch `size` and `hash`), which I found much worse than dropping them from the output